File: app/src/main/java/meow/softer/mydiary/entries/BaseDiaryFragment.java
package meow.softer.mydiary.entries;


import androidx.fragment.app.Fragment;

import meow.softer.mydiary.entries.entry.EntriesEntity;

import java.util.List;

public class BaseDiaryFragment extends Fragment {
    public long getTopicId() {

        return ((DiaryActivity) getActivity()).getTopicId();

    }
    public List<EntriesEntity> getEntriesList() {
        return ((DiaryActivity) getActivity()).getEntriesList();
    }

    public void updateEntriesList() {
        ((DiaryActivity) getActivity()).loadEntries();
    }
}

File: app/src/main/java/meow/softer/mydiary/entries/BaseDiaryFragment.kt
package meow.softer.mydiary.entries

import androidx.fragment.app.Fragment
import meow.softer.mydiary.entries.entry.EntriesEntity


open class BaseDiaryFragment : Fragment() {
    val topicId: Long
        get() = (activity as DiaryActivity).topicId
    val entriesList: MutableList<EntriesEntity?>?
        get() = (activity as DiaryActivity).entriesList

    fun updateEntriesList() {
        (activity as DiaryActivity).loadEntries()
    }
}
